### PR TITLE
Fix ring_exchange RS to support CUDA graph capture

### PR DESCRIPTION
**Pull request by vasunvidia**

### CODE REVIEW

**Comment by ksivaman** [comment]
@vasunvidia Could you make the change to main and not directly to release branch. Let's also replace all instances of `at::cuda::getDefaultCUDAStream` in `com_gemm_overlap.h` to `at::cuda::getCurrentCUDAStream`?

File: transformer_engine/pytorch/csrc/comm_gemm_overlap.h [hunk 1]
```diff
@@ -1097,6 +1099,13 @@ struct UbufP2PCommOverlap : torch::CustomClassHolder, UbufBase {
                            _ub_comm, recv_rank, (cudaStream_t) _stream_recv);
       }
     }
+    for (int i = 0; i < _stream_compute.size(); i++) {
+      CHECK_CUDA(
+          cudaEventRecord(_stop_compute, (cudaStream_t)_stream_compute[i]));
+      CHECK_CUDA(cudaStreamWaitEvent((cudaStream_t)stream_main, _stop_compute, 0));
+    }
+    CHECK_CUDA(cudaEventRecord(_stop_send, (cudaStream_t)_stream_send));
```
[the remark this answers is not in the document]
Comment (vasunvidia) [reply transformer_engine/pytorch/csrc/comm_gemm_overlap.h:1107]: yes. This is requirement for CUDA graph. I'll add these at the end of the function call

File: transformer_engine/pytorch/csrc/comm_gemm_overlap.h [hunk 2]
```diff
@@ -1097,6 +1099,13 @@ struct UbufP2PCommOverlap : torch::CustomClassHolder, UbufBase {
...
+    CHECK_CUDA(cudaStreamWaitEvent((cudaStream_t)stream_main, _stop_send, 0));
```
[the remark this answers is not in the document]
Comment (vasunvidia) [reply transformer_engine/pytorch/csrc/comm_gemm_overlap.h:1108]: Requirement of CUDA graph is all streams that branch out of the capture stream should be waited by capture stream.

**Comment by vasunvidia** [comment]
> @vasunvidia Could you make the change to main and not directly to release branch. Let's also replace all instances of `at::cuda::getDefaultCUDAStream` in `com_gemm_overlap.h` to `at::cuda::getCurrentCUDAStream`?

Made the changes and created new PR https://github.com/NVIDIA/TransformerEngine/pull/811

**Pull request closed without merging**